File: 1547-destination-city/destination-city.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        unordered_set<string>set;
        unordered_map<string,int>map;

        for(auto it : paths){
            auto v1=it[0];
            auto v2=it[1];
            map[v1]++;
            if(set.find(v1)==set.end()){
                set.insert(v1);
            }
            if(set.find(v2)==set.end()){
                set.insert(v2);
            }
        }
        string ans="";
        for(auto it:set){
            if(map.find(it)==map.end()){
                ans=it;
                break;
            }
        }
        return ans;
    }
};

File: 1547-destination-city/destination-city.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Map<String,Integer>set=new HashMap<>();
         Map<String,Integer>map=new HashMap<>();

        for(List<String> it : paths){
            String v1=it.get(0);
            String v2=it.get(1);
            if(map.containsKey(v1)==false){
                map.put(v1,1);
            }
            if(set.containsKey(v1)==false){
                set.put(v1,1);
            }
            if(set.containsKey(v2)==false){
                 set.put(v2,1);
            }
        }
        String ans="";
        for(Map.Entry<String,Integer>it:set.entrySet()){
            if(map.containsKey(it.getKey())==false){
                ans=it.getKey();
                break;
            }
        }
        return ans;
    }
}